surch/organization.py

```diff
@@ -19,199 +19,200 @@
+            self.item_type,
+            self.organization,
+            'public',
+            repos_per_page,
+            page_num), auth=self.git_credentials)
+        return response.json()
+
+    def _parse_repo_data(self, repo_data):
+        return [dict((key, data[key]) for key in ['name', 'clone_url'])
+                for data in repo_data]
+
+    def _get_repos_data(self, repos_per_page=100):
+        self.logger.info(
+            'Retrieving repository information for the {0}...'.format(
+                'organization' if self.is_organization else 'user'))
+        org_data = self._get_org_data()
+        repo_count = org_data['public_repos']
+        last_page_number = repo_count / repos_per_page
+        if (repo_count % repos_per_page) > 0:
             # Adding 2 because 1 for the extra repos that mean more page,
             #  and 1 for the next for loop.
             last_page_number += 2
-
+            repos_data = []
             for page_num in range(1, last_page_number):
-                repository_data = requests.get(
-                    REPO_DETAILS_API_URL.format(self.item_type,
-                                                self.organization,
-                                                repository_type,
-                                                repository_per_page,
-                                                page_num), auth=auth)
-
-                for repository in repository_data.json():
-                    self.repository_data.append(repository)
-                self.repository_specific_data = \
-                    self._parse_json_list_of_dict(['name', url_type])
-
-    def _parse_json_list_of_dict(self, list_of_arguments):
-        return [dict((key, data[key]) for key in list_of_arguments)
-                for data in self.repository_data]
-
-    def search(self, search_list, url_type='clone_url'):
+                repo_data = self._get_repos_list(repos_per_page, page_num)
+                repos_data.extend(self._parse_repo_data(repo_data))
+            return repos_data
+
+    def get_include_list(self, repos_data, include=None, exclude=None):
+        repo_url_list = []
+        if include:
+            for repo_name in include:
+                for repo_data in repos_data:
+                    if repo_data['name'] == repo_name:
+                        repo_url_list.append(repo_data['clone_url'])
+        elif exclude:
+            for repo_data in repos_data:
+                if repo_data['name'] not in exclude:
+                    repo_url_list.append(repo_data['clone_url'])
+        else:
+            for repo_data in repos_data:
+                repo_url_list.append(repo_data['clone_url'])
+        return repo_url_list
+
+    def search(self, search_list):
         search_list = search_list or self.search_list
         if len(search_list) == 0:
-            lgr.error('You must supply at least one string to search for.')
+            self.logger.error(
```

Comment:
if search_list can be None then you need to handle this since len(None) does not exist.